jobs/runner.go

```diff
@@ -0,0 +1,152 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobs
+
+import (
+	"context"
+	"runtime/pprof"
+	"sync"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm-agent/client/channel"
+)
+
+const jobsBufferSize = 32
+
+// Sender provides method for sending message to PMM server.
+type Sender interface {
+	SendResponse(msg *channel.AgentResponse)
+}
+
+// Runner allows to execute jobs.
+type Runner struct {
+	l *logrus.Entry
+
+	sender Sender
```

Comment:
let's use channels instead of sender interface, it'll bring us more issues with race conditions.

Reply:
calling sendResponse makes data flow uncontrollable

Reply:
And harder to read and understand

Reply:
Done